cmd/control/control.go

```diff
@@ -332,13 +389,33 @@ func startFunc(*cobra.Command, []string) error {
 		client.WithRootDataDir(rootDataDir),
 		client.WithReassignPortsIfUsed(reassignPortsIfUsed),
 		client.WithDynamicPorts(dynamicPorts),
-		client.WithNetworkID(networkID),
 	}
 
+	if upgradePath != "" {
+		opts = append(opts, client.WithUpgradePath(upgradePath))
+	}
+
+	if customNetworkGenesisPath != "" {
```

Comment:
you can do this inside the other function

Reply:
wdym?

Reply:
it's done here because of the `networkID, err = utils.NetworkIDFromGenesis(genesisBytes)`

Reply:
done